include/zephyr/drivers/adc/current_sense_amplifier.h

```diff
@@ -12,9 +12,9 @@
 
 struct current_sense_amplifier_dt_spec {
 	const struct adc_dt_spec port;
-	uint32_t sense_micro_ohms;
-	uint32_t sense_gain_mult;
-	uint32_t sense_gain_div;
+	uint32_t sense_milli_ohms;
+	uint16_t sense_gain_mult;
+	uint16_t sense_gain_div;
```

Comment:
need to update the DT binding description saying what the new maximum value is (before, the maximum was implied by the fact that a DT cell is 32 bits), and probably update the release notes mentioning it as well. might even want to do a compile time check of the property from DT to make sure it is not too large.

Reply:
Added the new limits to the migration guide and the dts bindings.
Compile time checks of the property in `CURRENT_SENSE_AMPLIFIER_DT_SPEC_GET` are not possible due to its position inside a struct initializer.

Reply:
It is possible, use BUILD_ASSERT

Reply:
Have you actually tried it yourself? As I previously stated, its not possible in a struct initializer as far as I can tell.
```
#define CURRENT_SENSE_AMPLIFIER_DT_SPEC_GET(node_id)                                               \
	{                                                                                          \
		BUILD_ASSERT(1, "sdf"),                                                            \
...
```
```
from /home/jordan/code/workspace/zephyr/drivers/sensor/current_amp/current_amp.c:9:
/home/jordan/code/workspace/zephyr/include/zephyr/toolchain/gcc.h:87:36: error: expected expression before '_Static_assert'
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert((EXPR), "" MSG)
```

Reply:
How about defining a macro like `CURRENT_SENSE_AMPLIFIER_DT_SPEC_VALIDATE(node_id)` that would contain `BUILD_ASSERT(DT_PROP(node_id, sense_resistor_micro_ohms) >= 1000, <error message>)`? It could be used right after `CURRENT_SENSE_AMPLIFIER_DT_SPEC_GET()` to catch invalid configurations, for example inside the `CURRENT_SENSE_AMPLIFIER_INIT(inst)` macro in `current_amp.c` instead of that added runtime assertion.

Reply:
Validating `sense-resistor-micro-ohms` is a different point than what @decsny asked for (Not sure why that validation is required in the first place, as assigning a value larger than `UINT16_MAX` will raise compiler warnings anyway).

If you are worried about validating `sense-resistor-micro-ohms` beyond the runtime assertion I added, perhaps renaming the property as `sense-resistor-milli-ohms` is a simpler solution. Everyone gets a dts error once, leaving no questions about rounding/validation.

Reply:
I clearly said validate the property from DT which at least @anangl understands what I meant, I'm not sure why you would say I'm not asking for the thing I explicitly said and instead assume I meant something impossible

But you're right anyways that there should be an overflow warning, so it is a moot point, nevermind